File: app/src/main/java/com/juniperphoton/myerlist/presenter/CustomCategoryContract.java
package com.juniperphoton.myerlist.presenter;


import com.juniperphoton.myerlist.model.ToDoCategory;

import java.util.List;

public class CustomCategoryContract {
    public interface View {
        void initData(List<ToDoCategory> data);

        void showDialog();

        void hideDialog(int delay);

        void hideKeyboard();

        void finish();
    }

    public interface Presenter extends com.juniperphoton.myerlist.presenter.Presenter {
        void cancel();

        void commit();

        void refreshData();
    }
}

File: app/src/main/java/com/juniperphoton/myerlist/presenter/CustomCategoryContract.kt
package com.juniperphoton.myerlist.presenter

import com.juniperphoton.myerlist.model.ToDoCategory

class CustomCategoryContract {
    interface View {
        fun initData(data: MutableList<ToDoCategory>)

        fun showDialog()

        fun hideDialog(delayMillis: Long)

        fun postDelay(block: Runnable, delayMillis: Long)

        fun hideKeyboard()

        fun finish()
    }

    interface Presenter : com.juniperphoton.myerlist.presenter.Presenter {
        fun commit()

        fun refreshData()
    }
}